packages/api/src/cronJobs.ts

```diff
@@ -0,0 +1,118 @@
+import { EntityManager } from '@mikro-orm/core';
+import { PostgreSqlDriver } from '@mikro-orm/postgresql';
+import { Contribution } from './entities/Contribution';
+import { Project } from './entities/Project';
+import { User } from './entities/User';
+import { env } from './env';
+import logger from './logger';
+import { buildQueryBodyString } from './utils/github/buildContributionQuery';
+import { buildDateQuery } from './utils/github/buildDateQuery';
+import { buildProjectsQuery } from './utils/github/buildProjectsQuery';
+import { buildUsersQuery } from './utils/github/buildUserQuery';
+
+interface ContributionResponse {
+  data: {
+    search: {
+      pageInfo: {
+        hasNextPage: boolean;
+        endCursor: string | null;
+      };
+      nodes: PullRequestContribution[];
+    };
+  };
+}
+
+export interface PullRequestContribution {
+  id: string;
+  title: string;
+  permalink: string;
+  mergedAt: string;
+  author: {
+    login: string;
+  };
+}
+
+export const contributionPoll = async (entityManager: EntityManager<PostgreSqlDriver>) => {
+  try {
+    const timeRange = new Date();
+    const today = new Date();
+    const yesterday = new Date(today);
+    yesterday.setDate(today.getDate() - 100);
+    timeRange.setHours(timeRange.getHours() - 5);
+
+    const userList = await entityManager.find(User, {
+      $or: [
+        {
+          contributionsLastCheckedAt: {
+            $lt: timeRange,
+          },
+        },
+        {
+          contributionsLastCheckedAt: {
+            $eq: null,
+          },
+        },
+      ],
+    });
+
+    if (userList.length === 0) {
+      return;
+    }
+
+    const projectList = await entityManager.find(Project, {});
+    const projectsString = await buildProjectsQuery(projectList);
+    const dateString = buildDateQuery(yesterday, today);
```

Comment:
If we do that we will have to make call to github on a per user basis which will drastically increase how close we get to our rate limit.  Currently we have a rate limit of 5000 points to use per hour.  If we do the search in one big chunk, each call is cost 1 point and returns 100 nodes.  So if we can get a little less than 500,000 nodes per hour (less because we also do a call to get projects once).  If we do a call per user each call will still be 1 point, but we will return less nodes per call and make more calls reducing the amount of potential nodes we can get per hour.  This will get worse when we add in other contribution types in the future.

A compromise to the use a date range that uses the oldest lastChecked time as the base for the search.  This will cause us to potentially pick up more nodes the we need, but our it should still be better than the per user option.

Rate Limit Explination: https://docs.github.com/en/graphql/overview/resource-limitations